### PR TITLE
JBPM-8615 - Generic Service Task - SLA Due Date property

### PR DESCRIPTION
**Pull request by Josephblt**
https://issues.jboss.org/browse/JBPM-8615

SLA Due Date property is available for generic service task nodes.

### CODE REVIEW

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins please retest this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/resources/org/kie/workbench/common/stunner/bpmn/resources/i18n/StunnerBPMNConstants.properties [hunk 1]
```diff
@@ -176,6 +176,7 @@ org.kie.workbench.common.stunner.bpmn.definition.property.reassignment.Reassignm
 #######################
 org.kie.workbench.common.stunner.bpmn.definition.GenericServiceTask.title=Service Task
 org.kie.workbench.common.stunner.bpmn.definition.GenericServiceTask.description=Service Task
+org.kie.workbench.common.stunner.bpmn.definition.property.service.GenericServiceTaskExecutionSet.label=Implementation/Execution
```
Comment (wmedvede) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/resources/org/kie/workbench/common/stunner/bpmn/resources/i18n/StunnerBPMNConstants.properties]: good catch

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java [hunk 2]
```diff
@@ -36,13 +41,26 @@ public void testHashCode() {
...
     public void testEquals() {
         GenericServiceTaskExecutionSet a = new GenericServiceTaskExecutionSet();
         GenericServiceTaskExecutionSet b = new GenericServiceTaskExecutionSet();
         Assert.assertTrue(a.equals(b));
+
+        GenericServiceTaskExecutionSet c = new GenericServiceTaskExecutionSet(new GenericServiceTaskInfo(),
+                                                                              new SLADueDate(SLA_DUE_DATE_1));
+        GenericServiceTaskExecutionSet d = new GenericServiceTaskExecutionSet(new GenericServiceTaskInfo(),
+                                                                              new SLADueDate(SLA_DUE_DATE_1));
+        Assert.assertTrue(c.equals(d));
```
Comment (wmedvede) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java]: Assert.assertTrue is used here, and we have a static import of  org.junit.Assert.assertEquals. I could probably make sense to unify criteria and have also a static import for assertEquals method.
(this is not a must, just a comment)

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java [hunk 3]
```diff
@@ -53,6 +71,12 @@ public void testEqualFalse() {
         a.setGenericServiceTaskInfo(null);
 
         Assert.assertFalse(a.equals(b));
```
Comment (wmedvede) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java]: same here

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/tasks/BaseTaskConverter.java [hunk 4]
```diff
@@ -142,7 +143,8 @@ BpmnNode bpmnServiceTask(org.eclipse.bpmn2.ServiceTask task) {
         ));
 
         definition.setExecutionSet(new GenericServiceTaskExecutionSet(
-                new GenericServiceTaskInfo(p.getGenericServiceTask())
+                new GenericServiceTaskInfo(p.getGenericServiceTask()),
+                new SLADueDate(p.getSlaDueDate())
```
Comment (wmedvede) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/tasks/BaseTaskConverter.java]: BaseTaskConverterTest is not updated to include this change

**Comment by wmedvede** [comment]
full downstream build failed

22:57:31 [WARNING] The requested profile "wildfly" could not be activated because it does not exist.
22:57:31 [WARNING] The requested profile "sourcemaps" could not be activated because it does not exist.
22:57:31 [WARNING] The requested profile "no-showcase" could not be activated because it does not exist.
22:57:31 [ERROR] Failed to execute goal on project optaplanner-wb-domain-editor-backend: Could not resolve dependencies for project org.optaplanner:optaplanner-wb-domain-editor-backend:jar:7.24.0-SNAPSHOT: Could not find artifact org.drools:drools-wb-globals-editor-api:jar:7.24.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/) -> [Help 1]
22:57:31 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project optaplanner-wb-domain-editor-backend: Could not resolve dependencies for project org.optaplanner:optaplanner-wb-domain-editor-backend:jar:7.24.0-SNAPSHOT: Could not find artifact org.drools:drools-wb-globals-editor-api:jar:7.24.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/)
22:57:31     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies (LifecycleDependencyResolver.java:249)
22:57:31     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:145)
22:57:31     at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:246)
22:57:31     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
22:57:31     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
22:57:31     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)

**Comment by wmedvede** [comment]
jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java [hunk 5]
```diff
@@ -61,11 +85,26 @@ public void testEqualTrue() {
...
         GenericServiceTaskExecutionSet a = new GenericServiceTaskExecutionSet();
         Assert.assertTrue(a.getGenericServiceTaskInfo().equals(new GenericServiceTaskInfo()));
     }
+
+    @Test
+    public void testGetSlaDueDate() {
+        GenericServiceTaskExecutionSet a = new GenericServiceTaskExecutionSet(
+                new GenericServiceTaskInfo(),
+                new SLADueDate(SLA_DUE_DATE_1));
+        Assert.assertTrue(a.getSlaDueDate().getValue().equals(SLA_DUE_DATE_1));
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java]: You can use assertEquals here to express what is an expected value.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by wmedvede** [comment]
build failed because of errors below, I think this should be fixed now in master

16:42:44 [ERROR] 
16:42:44 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
16:42:44 [ERROR] 
16:42:44 [ERROR] For more information about the errors and possible solutions, please read the following articles:
16:42:44 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
16:42:44 [ERROR] 
16:42:44 [ERROR] After correcting the problems, you can resume the build with the command
16:42:44 [ERROR]   mvn <goals> -rf :kie-wb-common-compiler-distribution

**Comment by wmedvede** [comment]
jenkins retest this

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Hello @LuboTerifaj, @wmedvede. All good now, 4 failed tests, but the wars were  generated and the tests are not related to this PR. I think this is good to test. Thanks.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java [hunk 6]
```diff
@@ -54,57 +53,55 @@ public void testHashCode() {
...
+        assertNotEquals(c, d);
     }
 
     @Test
     public void testEqualTrue() {
         GenericServiceTaskExecutionSet a = new GenericServiceTaskExecutionSet();
         GenericServiceTaskExecutionSet b = new GenericServiceTaskExecutionSet();
         a.setGenericServiceTaskInfo(new GenericServiceTaskInfo());
-        Assert.assertTrue(a.equals(b));
+        assertEquals(a, b);
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java]: Same as above.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java [hunk 7]
```diff
@@ -54,57 +53,55 @@ public void testHashCode() {
...
 
         GenericServiceTaskExecutionSet c = new GenericServiceTaskExecutionSet(new GenericServiceTaskInfo(),
                                                                               new SLADueDate(SLA_DUE_DATE_1));
         GenericServiceTaskExecutionSet d = new GenericServiceTaskExecutionSet(new GenericServiceTaskInfo(),
                                                                               new SLADueDate(SLA_DUE_DATE_1));
-
-        Assert.assertTrue(c.equals(d));
+        assertEquals(c, d);
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java]: Same as above.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java [hunk 8]
```diff
@@ -54,57 +53,55 @@ public void testHashCode() {
     public void testEquals() {
         GenericServiceTaskExecutionSet a = new GenericServiceTaskExecutionSet();
         GenericServiceTaskExecutionSet b = new GenericServiceTaskExecutionSet();
-        Assert.assertTrue(a.equals(b));
+        assertEquals(a, b);
```
[the remark this answers is not in the document]
Comment (Josephblt) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/property/service/GenericServiceTaskExecutionSetTest.java]: This is just my opinion, but I think it drives intent. No real difference here other than the test flow "narrative", it just seems more clear to me.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDirectDiagramMarshallerTest.java [hunk 9]
```diff
@@ -227,6 +227,7 @@
     private static final String BPMN_LOG_TASK_JBPM_DESIGNER = "org/kie/workbench/common/stunner/bpmn/backend/service/diagram/logtask.bpmn";
     private static final String BPMN_SERVICETASKS_JBPM_DESIGNER = "org/kie/workbench/common/stunner/bpmn/backend/service/diagram/serviceTasksJBPMDeginer.bpmn";
     private static final String BPMN_EVENT_GATEWAY = "org/kie/workbench/common/stunner/bpmn/backend/service/diagram/eventGateway.bpmn";
+    private static final String BPMN_GENERIC_SERVICE_TASK = "org/kie/workbench/common/stunner/bpmn/backend/service/diagram/genericServiceTasks.bpmn";
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDirectDiagramMarshallerTest.java]: The constant is never used.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Hey @wmedvede @LuboTerifaj let's please try to move this forward, the builds are successfull. Thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDirectDiagramMarshallerTest.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (Josephblt) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDirectDiagramMarshallerTest.java]: Yeah, this was not used at all. This file is used in another test class (GenericServiceTaskTest) so I removed the line.

**Comment by Josephblt** [comment]
Hello @LuboTerifaj. Just missing your review here. Removed the unused constants. Thanks

**Comment by Josephblt** [comment]
Hey @romartin. All set here, ready to merge. Thanks.

**Pull request merged**